vortex-datafusion/src/persistent/opener.rs

```diff
@@ -424,6 +429,29 @@ fn byte_range_to_row_range(byte_range: Range<u64>, row_count: u64, total_size: u
     start_row..u64::min(row_count, end_row)
 }
 
+/// Attempts to extract a `Selection` from the extensions object, if present.
+///
+/// This function is used to retrieve the row selection plan that may have been
+/// attached to a `PartitionedFile` via its `extensions` field.
+///
+/// # Arguments
+///
+/// * `extensions` - Optional type-erased extensions object that may contain a `Selection`
+///
+/// # Returns
+///
+/// Returns `Some(Selection)` if the extensions contain a valid `Selection`, otherwise `None`.
+fn get_selection_from_extensions(
+    extensions: Option<Arc<dyn std::any::Any + Send + Sync>>,
+) -> Option<Selection> {
+    if let Some(extensions) = extensions
+        && let Some(selection) = extensions.downcast_ref::<Selection>()
+    {
+        return Some(selection.clone());
+    }
+    None
```

Comment:
You can simplify this:

```suggestion
    extensions?.downcast_ref::<Selection>().cloned()
```